tests/permissions.py

```diff
@@ -0,0 +1,1144 @@
+    field_id('Dataset', 'locations'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Dataset', 'owners'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Dataset', 'stack'): TestData(resource_perm=GET_DATASET, tenant_ignore=IgnoreReason.NOTREQUIRED),
+    field_id('Dataset', 'statistics'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Dataset', 'stewards'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Dataset', 'tables'): TestData(
+        resource_ignore=IgnoreReason.CUSTOM, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Dataset', 'terms'): TestData(resource_ignore=IgnoreReason.PUBLIC, tenant_ignore=IgnoreReason.NOTREQUIRED),
+    field_id('Dataset', 'userRoleForDataset'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetBase', 'environment'): TestData(
+        resource_perm=GET_ENVIRONMENT, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetBase', 'owners'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetBase', 'stack'): TestData(resource_perm=GET_DATASET, tenant_ignore=IgnoreReason.NOTREQUIRED),
+    field_id('DatasetBase', 'stewards'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetBase', 'userRoleForDataset'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetProfilingRun', 'dataset'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetProfilingRun', 'results'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetProfilingRun', 'status'): TestData(
+        resource_ignore=IgnoreReason.INTRAMODULE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetStorageLocation', 'dataset'): TestData(
+        resource_perm=GET_DATASET, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetStorageLocation', 'terms'): TestData(
+        resource_ignore=IgnoreReason.PUBLIC, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetTable', 'GlueTableProperties'): TestData(
+        resource_perm=GET_DATASET_TABLE, tenant_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('DatasetTable', 'columns'): TestData(
+        resource_ignore=IgnoreReason.CUSTOM, tenant_ignore=IgnoreReason.USERLIMITED
```

Comment:
`CUSTOM` as I understand it is that permissions are checked, but they are checked through their own mechanisms. Are there any checks happening in the columns resolver?

Reply:
Yes I changed this after @noah-paige suggestion because it is checking the Confidentiality Classification...
```
    @staticmethod
    def paginate_active_columns_for_table(uri: str, filter=None):
        context = get_context()
        with context.db_engine.scoped_session() as session:
            table: DatasetTable = DatasetTableRepository.get_dataset_table_by_uri(session, uri)
            dataset = DatasetRepository.get_dataset_by_uri(session, table.datasetUri)
            if (
                ConfidentialityClassification.get_confidentiality_level(dataset.confidentiality)
                != ConfidentialityClassification.Unclassified.value
            ) and (dataset.SamlAdminGroupName not in context.groups and dataset.stewards not in context.groups):
                raise exceptions.UnauthorizedOperation(
                    action='LIST_DATASET_TABLE_COLUMNS',
                    message='User is not authorized to view Columns for Confidential datasets',
                )
            return DatasetColumnRepository.paginate_active_columns_for_table(session, uri, filter)
```